pennylane/measurements/expval.py

```diff
@@ -62,6 +62,12 @@ def circuit(x):
             "qml.expval does not support measuring sequences of measurements or observables"
         )
 
+    if isinstance(op, qml.Identity) and len(op.wires) == 0:
+        # temporary solution to merge https://github.com/PennyLaneAI/pennylane/pull/5106
+        raise NotImplementedError(
+            "Expectation values of qml.Identity() without wires are currently not allowed."
```

Comment:
Not entirely sure if this is what you had in mind @albi3ro for a temporary solution, I am a bit worried about performance implications since this will affect a vast majority of users 😬 I am wondering if it may not be better to not do this and leave as fix later.